# Análise de Dados - Prouni 2015

Conjunto de dados que apresenta as bolsas concedidas e o perfil dos beneficiários do ProUni de 2015.

Os dados são disponibilizados pelo Site Dados Abertos (http://dadosabertos.mec.gov.br/prouni), em arquivo CSV.

In [1]:
import pandas as pd

df = pd.read_csv ('datasets/pda-prouni-2015.csv', sep = ';', encoding = 'ISO-8859-1')

## Conhecendo o dataframe

In [2]:
df.sample(5)

ANO_CONCESSAO_BOLSA  CODIGO_EMEC_IES_BOLSA  \
200435                 2015                    802   
226137                 2015                    466   
4327                   2015                    437   
94650                  2015                   3603   
177865                 2015                  10588   

                                          NOME_IES_BOLSA      TIPO_BOLSA  \
200435                      CENTRO UNIVERSITÁRIO CARIOCA  BOLSA INTEGRAL   
226137                      UNIVERSIDADE ANHEMBI MORUMBI  BOLSA INTEGRAL   
4327                             UNIVERSIDADE PARANAENSE  BOLSA INTEGRAL   
94650                  FACULDADE ANHANGUERA DE GUARULHOS  BOLSA INTEGRAL   
177865  FACULDADE JOAQUIM NABUCO DE SÃO LOURENÇO DA MATA  BOLSA INTEGRAL   

       MODALIDADE_ENSINO_BOLSA          NOME_CURSO_BOLSA  \
200435              PRESENCIAL  Engenharia De Computação   
226137              PRESENCIAL    Comércio Internacional   
4327                PRESENCIAL             Administração   
94650               PRESENCIAL  Publicidade E Propaganda   
177865              PRESENCIAL             Administração   

       NOME_TURNO_CURSO_BOLSA CPF_BENEFICIARIO_BOLSA SEXO_BENEFICIARIO_BOLSA  \
200435               Matutino            ***435567**               Masculino   
226137               Matutino            ***136111**               Masculino   
4327                  Noturno            ***101949**               Masculino   
94650                 Noturno            ***690188**               Masculino   
177865               Matutino            ***894744**                Feminino   

       RACA_BENEFICIARIO_BOLSA DT_NASCIMENTO_BENEFICIARIO  \
200435                   Preta                 24-04-1996   
226137                   Parda                 20-07-1995   
4327                     Preta                 12-05-1982   
94650                    Parda                 21-11-1997   
177865                   Parda                 04-11-1996   

       BENEFICIARIO_DEFICIENTE_FISICO REGIAO_BENEFICIARIO_BOLSA  \
200435                            NÃO                   SUDESTE   
226137                            NÃO                   SUDESTE   
4327                              NÃO                       SUL   
94650                             NÃO                   SUDESTE   
177865                            NÃO                  NORDESTE   

       SIGLA_UF_BENEFICIARIO_BOLSA MUNICIPIO_BENEFICIARIO_BOLSA  
200435                          RJ                 BELFORD ROXO  
226137                          SP        SÃO BERNARDO DO CAMPO  
4327                            PR                     CASCAVEL  
94650                           SP                    GUARULHOS  
177865                          PE                   CAMARAGIBE

In [3]:
df.shape

(252650, 15)

In [4]:
df.dtypes

ANO_CONCESSAO_BOLSA                int64
CODIGO_EMEC_IES_BOLSA              int64
NOME_IES_BOLSA                    object
TIPO_BOLSA                        object
MODALIDADE_ENSINO_BOLSA           object
NOME_CURSO_BOLSA                  object
NOME_TURNO_CURSO_BOLSA            object
CPF_BENEFICIARIO_BOLSA            object
SEXO_BENEFICIARIO_BOLSA           object
RACA_BENEFICIARIO_BOLSA           object
DT_NASCIMENTO_BENEFICIARIO        object
BENEFICIARIO_DEFICIENTE_FISICO    object
REGIAO_BENEFICIARIO_BOLSA         object
SIGLA_UF_BENEFICIARIO_BOLSA       object
MUNICIPIO_BENEFICIARIO_BOLSA      object
dtype: object

In [5]:
df.isnull().sum()

ANO_CONCESSAO_BOLSA                 0
CODIGO_EMEC_IES_BOLSA               0
NOME_IES_BOLSA                      0
TIPO_BOLSA                          0
MODALIDADE_ENSINO_BOLSA             0
NOME_CURSO_BOLSA                  137
NOME_TURNO_CURSO_BOLSA              0
CPF_BENEFICIARIO_BOLSA              0
SEXO_BENEFICIARIO_BOLSA             0
RACA_BENEFICIARIO_BOLSA             0
DT_NASCIMENTO_BENEFICIARIO          0
BENEFICIARIO_DEFICIENTE_FISICO      0
REGIAO_BENEFICIARIO_BOLSA          15
SIGLA_UF_BENEFICIARIO_BOLSA        15
MUNICIPIO_BENEFICIARIO_BOLSA       15
dtype: int64

## Dados nulos

- A Coluna NOME_CURSO_BOLSA, consta a falta de 137 registros;


- As Colunas REGIAO_BENEFICIARIO_BOLSA , SIGLA_UF_BENEFICIARIO_BOLSA e MUNICIPIO_BENEFICIARIO_BOLSA, constam falta de 15 registros.

### Atribuindo valores na Coluna NOME_CURSO_BOLSA

-> Todos os elementos vazios serão preenchidos com o valor definido 'Não Registrado'

In [6]:
df[df['NOME_CURSO_BOLSA'].isnull()]

ANO_CONCESSAO_BOLSA  CODIGO_EMEC_IES_BOLSA  \
1427                   2015                    322   
3269                   2015                    163   
3408                   2015                    163   
5470                   2015                    163   
6312                   2015                    437   
...                     ...                    ...   
242044                 2015                   3772   
242382                 2015                   3772   
246184                 2015                     83   
247305                 2015                   2271   
247340                 2015                    472   

                                           NOME_IES_BOLSA      TIPO_BOLSA  \
1427                                UNIVERSIDADE PAULISTA  BOLSA INTEGRAL   
3269                           UNIVERSIDADE ESTÁCIO DE SÁ  BOLSA INTEGRAL   
3408                           UNIVERSIDADE ESTÁCIO DE SÁ  BOLSA INTEGRAL   
5470                           UNIVERSIDADE ESTÁCIO DE SÁ  BOLSA INTEGRAL   
6312                              UNIVERSIDADE PARANAENSE  BOLSA INTEGRAL   
...                                                   ...             ...   
242044                                    Faculdade IBGEN  BOLSA INTEGRAL   
242382                                    Faculdade IBGEN  BOLSA INTEGRAL   
246184                     UNIVERSIDADE DO VALE DO ITAJAÍ  BOLSA INTEGRAL   
247305                    FACULDADE PITÁGORAS DE IPATINGA  BOLSA INTEGRAL   
247340  UNIVERSIDADE DO GRANDE RIO PROFESSOR JOSÉ DE S...  BOLSA INTEGRAL   

       MODALIDADE_ENSINO_BOLSA NOME_CURSO_BOLSA NOME_TURNO_CURSO_BOLSA  \
1427                PRESENCIAL              NaN                Noturno   
3269      EDUCAÇÃO A DISTÂNCIA              NaN      Curso a distância   
3408                PRESENCIAL              NaN                Noturno   
5470                PRESENCIAL              NaN                Noturno   
6312                PRESENCIAL              NaN                Noturno   
...                        ...              ...                    ...   
242044              PRESENCIAL              NaN               Integral   
242382              PRESENCIAL              NaN               Integral   
246184              PRESENCIAL              NaN                Noturno   
247305              PRESENCIAL              NaN               Matutino   
247340              PRESENCIAL              NaN               Matutino   

       CPF_BENEFICIARIO_BOLSA SEXO_BENEFICIARIO_BOLSA RACA_BENEFICIARIO_BOLSA  \
1427              ***398105**               Masculino                  Branca   
3269              ***601867**               Masculino                   Parda   
3408              ***074207**               Masculino           Não Informada   
5470              ***649357**                Feminino                  Branca   
6312              ***715139**                Feminino                  Branca   
...                       ...                     ...                     ...   
242044            ***939780**                Feminino                  Branca   
242382            ***790010**               Masculino                  Branca   
246184            ***656159**                Feminino                  Branca   
247305            ***599136**                Feminino                  Branca   
247340            ***141767**                Feminino                  Branca   

       DT_NASCIMENTO_BENEFICIARIO BENEFICIARIO_DEFICIENTE_FISICO  \
1427                   02-09-1984                            NÃO   
3269                   20-06-1991                            NÃO   
3408                   30-12-1992                            NÃO   
5470                   27-03-1998                            NÃO   
6312                   23-12-1997                            NÃO   
...                           ...                            ...   
242044                 10-07-1990                            NÃO   
242382                 02-12

In [7]:
df.update(df['NOME_CURSO_BOLSA'].fillna('Não Registrado'))
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 252650 entries, 0 to 252649
Data columns (total 15 columns):
 #   Column                          Non-Null Count   Dtype 
---  ------                          --------------   ----- 
 0   ANO_CONCESSAO_BOLSA             252650 non-null  int64 
 1   CODIGO_EMEC_IES_BOLSA           252650 non-null  int64 
 2   NOME_IES_BOLSA                  252650 non-null  object
 3   TIPO_BOLSA                      252650 non-null  object
 4   MODALIDADE_ENSINO_BOLSA         252650 non-null  object
 5   NOME_CURSO_BOLSA                252650 non-null  object
 6   NOME_TURNO_CURSO_BOLSA          252650 non-null  object
 7   CPF_BENEFICIARIO_BOLSA          252650 non-null  object
 8   SEXO_BENEFICIARIO_BOLSA         252650 non-null  object
 9   RACA_BENEFICIARIO_BOLSA         252650 non-null  object
 10  DT_NASCIMENTO_BENEFICIARIO      252650 non-null  object
 11  BENEFICIARIO_DEFICIENTE_FISICO  252650 non-null  object
 12  REGIAO_BENEFICIARIO_BOLSA     

### Analisando as Colunas:

REGIAO_BENEFICIARIO_BOLSA, SIGLA_UF_BENEFICIARIO_BOLSA e MUNICIPIO_BENEFICIARIO_BOLSA


- 1º: será atribuído o valor 'Não Cadastrado' nas linhas vazias de 3 colunas;


- 2º: Após a análise do df e com o intuito de minimizar esses registros, vê-se a possibilidade de substituição dos valores nas colunas: REGIAO_BENEFICIARIO_BOLSA e SIGLA_UF_BENEFICIARIO_BOLSA

-> essa substituição terá como base os Código das IE's ou os Nomes das IE's

-> a substituição ocorrerá somente para as IE's mais frequentes no dataframe


In [8]:
df['REGIAO_BENEFICIARIO_BOLSA'].fillna('Não Registrada', inplace = True) 
df['SIGLA_UF_BENEFICIARIO_BOLSA'].fillna('Não Registrada', inplace = True) 
df['MUNICIPIO_BENEFICIARIO_BOLSA'].fillna('Não Registrado', inplace = True)

In [9]:
df.loc[(df['REGIAO_BENEFICIARIO_BOLSA'] == 'Não Registrada')].head(15)

ANO_CONCESSAO_BOLSA  CODIGO_EMEC_IES_BOLSA  \
3667                   2015                    308   
7778                   2015                    457   
26935                  2015                    203   
27456                  2015                    203   
56284                  2015                    203   
57316                  2015                    203   
67035                  2015                    517   
85425                  2015                   1645   
91589                  2015                   2040   
95491                  2015                   2497   
98280                  2015                   4613   
100011                 2015                    522   
128002                 2015                   2917   
130657                 2015                   4362   
141172                 2015                    673   

                                         NOME_IES_BOLSA         TIPO_BOLSA  \
3667             UNIVERSIDADE PRESIDENTE ANTÔNIO CARLOS     BOLSA INTEGRAL   
7778    UNIVERSIDADE ANHANGUERA DE SÃO PAULO - UNIAN-SP     BOLSA INTEGRAL   
26935                      UNIVERSIDADE SÃO JUDAS TADEU  BOLSA PARCIAL 50%   
27456                      UNIVERSIDADE SÃO JUDAS TADEU  BOLSA PARCIAL 50%   
56284                      UNIVERSIDADE SÃO JUDAS TADEU  BOLSA PARCIAL 50%   
57316                      UNIVERSIDADE SÃO JUDAS TADEU  BOLSA PARCIAL 50%   
67035                    FACULDADE DE EDUCAÇÃO SÃO LUÍS     BOLSA INTEGRAL   
85425      FACULDADE DE TECNOLOGIA E CIÊNCIAS DE JEQUIÉ  BOLSA PARCIAL 50%   
91589                               FACULDADE DO FUTURO     BOLSA INTEGRAL   
95491                                  FACULDADE ATENEU     BOLSA INTEGRAL   
98280             FACULDADE DE EDUCAÇÃO E MEIO AMBIENTE     BOLSA INTEGRAL   
100011                CENTRO UNIVERSITÁRIO CELSO LISBOA     BOLSA INTEGRAL   
128002                           FACULDADE BRASIL NORTE     BOLSA INTEGRAL   
130657                     FACULDADE PITÁGORAS DE BETIM     BOLSA INTEGRAL   
141172          CENTRO UNIVERSITÁRIO DA GRANDE DOURADOS     BOLSA INTEGRAL   

       MODALIDADE_ENSINO_BOLSA            NOME_CURSO_BOLSA  \
3667                PRESENCIAL          Ciências Contábeis   
7778                PRESENCIAL             Educação Física   
26935               PRESENCIAL               Administração   
27456               PRESENCIAL      Sistemas De Informação   
56284               PRESENCIAL            Engenharia Civil   
57316               PRESENCIAL                 Biomedicina   
67035               PRESENCIAL                    História   
85425               PRESENCIAL            Engenharia Civil   
91589               PRESENCIAL                    Farmácia   
95491               PRESENCIAL  Gestão De Recursos Humanos   
98280               PRESENCIAL             Educação Física   
100011              PRESENCIAL                    Farmácia   
128002              PRESENCIAL                     Direito   
130657              PRESENCIAL               Administração   
141172    EDUCAÇÃO A DISTÂNCIA  Letras - Língua Portuguesa   

       NOME_TURNO_CURSO_BOLSA CPF_BENEFICIARIO_BOLSA SEXO_BENEFICIARIO_BOLSA  \
3667                  Noturno            ***615976**                Feminino   
7778                 Matutino            ***047388**               Masculino   
26935                 Noturno            ***072558**                Feminino   
27456                 Noturno            ***053912**               Masculino   
56284                 Noturno            ***685138**               Masculino   
57316                Matutino            ***512587**                Feminino   
67035                 Noturno            ***280388**               Masculino   
85425                 Noturno            ***461005**               Masculino   
91589                 Noturno            ***435176**                Feminino   
95491                Matutino            ***411883**               Masculino   
98280                

In [10]:
#df.loc[(df['CODIGO_EMEC_IES_BOLSA'] == )].count()

In [11]:
#df.loc[(df['CODIGO_EMEC_IES_BOLSA'] == )].sample(10)

### Substitundo os valores nas colunas REGIAO_BENEFICIARIO_BOLSA e SIGLA_UF_BENEFICIARIO_BOLSA

In [12]:
df.loc[(df ['CODIGO_EMEC_IES_BOLSA'] == 203) & (df ['REGIAO_BENEFICIARIO_BOLSA'] == 'Não Registrada'), 'REGIAO_BENEFICIARIO_BOLSA'] = 'SUDESTE'

In [13]:
df.loc[(df ['CODIGO_EMEC_IES_BOLSA'] == 203) & (df ['SIGLA_UF_BENEFICIARIO_BOLSA'] == 'Não Registrada'), 'SIGLA_UF_BENEFICIARIO_BOLSA'] = 'SP'

In [14]:
df.loc[(df['REGIAO_BENEFICIARIO_BOLSA'] == 'Não Registrada')].count()

ANO_CONCESSAO_BOLSA               11
CODIGO_EMEC_IES_BOLSA             11
NOME_IES_BOLSA                    11
TIPO_BOLSA                        11
MODALIDADE_ENSINO_BOLSA           11
NOME_CURSO_BOLSA                  11
NOME_TURNO_CURSO_BOLSA            11
CPF_BENEFICIARIO_BOLSA            11
SEXO_BENEFICIARIO_BOLSA           11
RACA_BENEFICIARIO_BOLSA           11
DT_NASCIMENTO_BENEFICIARIO        11
BENEFICIARIO_DEFICIENTE_FISICO    11
REGIAO_BENEFICIARIO_BOLSA         11
SIGLA_UF_BENEFICIARIO_BOLSA       11
MUNICIPIO_BENEFICIARIO_BOLSA      11
dtype: int64

### Excluindo as Colunas CPF_BENEFICIARIO_BOLSA e MUNICIPIO_BENEFICIARIO_BOLSA

In [15]:
df.drop(['CPF_BENEFICIARIO_BOLSA','MUNICIPIO_BENEFICIARIO_BOLSA'] , axis=1, inplace = True)

### Renomeando Colunas

In [16]:
df.columns

Index(['ANO_CONCESSAO_BOLSA', 'CODIGO_EMEC_IES_BOLSA', 'NOME_IES_BOLSA',
       'TIPO_BOLSA', 'MODALIDADE_ENSINO_BOLSA', 'NOME_CURSO_BOLSA',
       'NOME_TURNO_CURSO_BOLSA', 'SEXO_BENEFICIARIO_BOLSA',
       'RACA_BENEFICIARIO_BOLSA', 'DT_NASCIMENTO_BENEFICIARIO',
       'BENEFICIARIO_DEFICIENTE_FISICO', 'REGIAO_BENEFICIARIO_BOLSA',
       'SIGLA_UF_BENEFICIARIO_BOLSA'],
      dtype='object')

In [17]:
df.rename(columns = {'ANO_CONCESSAO_BOLSA':'ANO_CONCESSAO', 
                     'CODIGO_EMEC_IES_BOLSA':'CODIGO_IES', 
                     'NOME_IES_BOLSA':'NOME_IES', 
                     'MODALIDADE_ENSINO_BOLSA':'MODALIDADE_ENSINO', 
                     'NOME_CURSO_BOLSA':'NOME_CURSO'}, inplace = True)

In [18]:
df.rename(columns = {'NOME_TURNO_CURSO_BOLSA':'TURNO_CURSO', 
                     'SEXO_BENEFICIARIO_BOLSA':'SEXO_BENEFICIARIO', 
                     'RACA_BENEFICIARIO_BOLSA':'RACA_BENEFICIARIO', 
                     'REGIAO_BENEFICIARIO_BOLSA':'REGIAO_BENEFICIARIO', 
                     'SIGLA_UF_BENEFICIARIO_BOLSA':'UF_BENEFICIARIO', 
                     'DT_NASCIMENTO_BENEFICIARIO':'ANO_NASCIMENTO_BENEFICIARIO'}, inplace = True)

In [19]:
df.sample(5)

ANO_CONCESSAO  CODIGO_IES  \
214675           2015        1642   
68797            2015        1196   
187608           2015          14   
600              2015         494   
69833            2015        1058   

                                                 NOME_IES         TIPO_BOLSA  \
214675      FACULDADE DE TECNOLOGIA E CIÊNCIAS DE ITABUNA  BOLSA PARCIAL 50%   
68797        CENTRO UNIVERSITÁRIO DE MARINGÁ - UNICESUMAR     BOLSA INTEGRAL   
187608              UNIVERSIDADE DO VALE DO RIO DOS SINOS     BOLSA INTEGRAL   
600                 UNIVERSIDADE DO SUL DE SANTA CATARINA     BOLSA INTEGRAL   
69833   Estácio FIB - Centro Universitário Estácio da ...     BOLSA INTEGRAL   

       MODALIDADE_ENSINO           NOME_CURSO TURNO_CURSO SEXO_BENEFICIARIO  \
214675        PRESENCIAL           Psicologia     Noturno          Feminino   
68797         PRESENCIAL     Engenharia Civil    Matutino          Feminino   
187608        PRESENCIAL  Engenharia Mecânica     Noturno         Masculino   
600           PRESENCIAL              Direito     Noturno         Masculino   
69833         PRESENCIAL       Petróleo E Gás     Noturno         Masculino   

       RACA_BENEFICIARIO ANO_NASCIMENTO_BENEFICIARIO  \
214675             Parda                  02-01-1976   
68797              Parda                  03-11-1997   
187608            Branca                  24-12-1990   
600              Amarela                  26-06-1997   
69833              Preta                  17-05-1974   

       BENEFICIARIO_DEFICIENTE_FISICO REGIAO_BENEFICIARIO UF_BENEFICIARIO  
214675                            NÃO            NORDESTE              BA  
68797                             NÃO                 SUL              PR  
187608                            NÃO                 SUL              RS  
600                               NÃO                 SUL              SC  
69833                             NÃO            NORDESTE              BA

### Descobrindo a idade aproximada* dos beneficiários

* é aproximada, pois o dataframe não menciona qual semestre do ano o beneficiário foi contemplado pela bolsa.

-> Criar uma nova coluna IDADE_APROX_BENEFICIARIO para calcular a idade aproximada deles

-> Alterar o formato da coluna DT_NASCIMENTO_BENEFICIARIO Para Y

-> Calcular com base na coluna ANO_CONCESSAO a idade aproximada do beneficiário

In [20]:
df['ANO_NASCIMENTO_BENEFICIARIO'] = pd.to_datetime(df['ANO_NASCIMENTO_BENEFICIARIO']).dt.year
print(df['ANO_NASCIMENTO_BENEFICIARIO'])

0         1994
1         1986
2         1994
3         1993
4         1995
          ... 
252645    1986
252646    1982
252647    1995
252648    1996
252649    1995
Name: ANO_NASCIMENTO_BENEFICIARIO, Length: 252650, dtype: int64


In [21]:
df.insert(loc=10, column='IDADE_APROX_BENEFICIARIO', value=10)

In [22]:
df.head(1)

ANO_CONCESSAO  CODIGO_IES                     NOME_IES      TIPO_BOLSA  \
0           2015          20  UNIVERSIDADE DE PASSO FUNDO  BOLSA INTEGRAL   

  MODALIDADE_ENSINO     NOME_CURSO TURNO_CURSO SEXO_BENEFICIARIO  \
0        PRESENCIAL  Administração     Noturno          Feminino   

  RACA_BENEFICIARIO  ANO_NASCIMENTO_BENEFICIARIO  IDADE_APROX_BENEFICIARIO  \
0            Branca                         1994                        10   

  BENEFICIARIO_DEFICIENTE_FISICO REGIAO_BENEFICIARIO UF_BENEFICIARIO  
0                            NÃO                 SUL              RS

In [23]:
df['IDADE_APROX_BENEFICIARIO'] = df['ANO_CONCESSAO'] - df['ANO_NASCIMENTO_BENEFICIARIO']
df.sample(5)

ANO_CONCESSAO  CODIGO_IES                                 NOME_IES  \
50031            2015        3333             CENTRO UNIVERSITÁRIO UNIFTEC   
197577           2015         673  CENTRO UNIVERSITÁRIO DA GRANDE DOURADOS   
179689           2015         738      CENTRO UNIVERSITÁRIO TERESA D'ÁVILA   
132043           2015       14002       Faculdade dos Guararapes de Recife   
98828            2015         296       UNIVERSIDADE DA REGIÃO DA CAMPANHA   

               TIPO_BOLSA MODALIDADE_ENSINO                NOME_CURSO  \
50031      BOLSA INTEGRAL        PRESENCIAL  Engenharia De Computação   
197577  BOLSA PARCIAL 50%        PRESENCIAL     Ciência Da Computação   
179689     BOLSA INTEGRAL        PRESENCIAL   Arquitetura E Urbanismo   
132043     BOLSA INTEGRAL        PRESENCIAL          Engenharia Civil   
98828      BOLSA INTEGRAL        PRESENCIAL              Fisioterapia   

       TURNO_CURSO SEXO_BENEFICIARIO RACA_BENEFICIARIO  \
50031      Noturno          Feminino            Branca   
197577     Noturno         Masculino            Branca   
179689     Noturno          Feminino            Branca   
132043     Noturno          Feminino             Parda   
98828      Noturno          Feminino            Branca   

        ANO_NASCIMENTO_BENEFICIARIO  IDADE_APROX_BENEFICIARIO  \
50031                          1985                        30   
197577                         1996                        19   
179689                         1997                        18   
132043                         1993                        22   
98828                          1998                        17   

       BENEFICIARIO_DEFICIENTE_FISICO REGIAO_BENEFICIARIO UF_BENEFICIARIO  
50031                             NÃO                 SUL              RS  
197577                            NÃO        CENTRO-OESTE              MS  
179689                            NÃO             SUDESTE              SP  
132043                            NÃO            NORDESTE              PE  
98828                             NÃO                 SUL              RS

### Criando intervalos em que se agrupam pessoas de idade aproximada

-> Criar uma nova coluna FAIXA_ETA_BENEFICIARIO

-> Popular a coluna seguindo os seguintes critérios:

    Menos de 18 anos
    De 18 a 24 anos
    De 24 a 30 anos
    De 30 a 36 anos
    De 36 a 48 anos
    De 48 a 60 anos
    Mais de 60 anos

In [24]:
df.insert(loc=11, column='FAIXA_ETA_BENEFICIARIO', value='Idade')

In [25]:
df.head()

ANO_CONCESSAO  CODIGO_IES  \
0           2015          20   
1           2015          20   
2           2015          20   
3           2015         423   
4           2015         423   

                                            NOME_IES         TIPO_BOLSA  \
0                        UNIVERSIDADE DE PASSO FUNDO     BOLSA INTEGRAL   
1                        UNIVERSIDADE DE PASSO FUNDO  BOLSA PARCIAL 50%   
2                        UNIVERSIDADE DE PASSO FUNDO  BOLSA PARCIAL 50%   
3  UNIVERSIDADE REGIONAL INTEGRADA DO ALTO URUGUA...     BOLSA INTEGRAL   
4  UNIVERSIDADE REGIONAL INTEGRADA DO ALTO URUGUA...  BOLSA PARCIAL 50%   

  MODALIDADE_ENSINO          NOME_CURSO TURNO_CURSO SEXO_BENEFICIARIO  \
0        PRESENCIAL       Administração     Noturno          Feminino   
1        PRESENCIAL       Administração     Noturno          Feminino   
2        PRESENCIAL       Administração     Noturno          Feminino   
3        PRESENCIAL  Ciências Contábeis     Noturno         Masculino   
4        PRESENCIAL  Ciências Contábeis     Noturno          Feminino   

  RACA_BENEFICIARIO  ANO_NASCIMENTO_BENEFICIARIO  IDADE_APROX_BENEFICIARIO  \
0            Branca                         1994                        21   
1            Branca                         1986                        29   
2            Branca                         1994                        21   
3            Branca                         1993                        22   
4            Branca                         1995                        20   

  FAIXA_ETA_BENEFICIARIO BENEFICIARIO_DEFICIENTE_FISICO REGIAO_BENEFICIARIO  \
0                  Idade                            NÃO                 SUL   
1                  Idade                            SIM                 SUL   
2                  Idade                            NÃO                 SUL   
3                  Idade                            NÃO                 SUL   
4                  Idade                            NÃO                 SUL   

  UF_BENEFICIARIO  
0              RS  
1              RS  
2              RS  
3              RS  
4              RS

In [26]:
for i in range(len (df)): 
    if (df.loc[i, 'IDADE_APROX_BENEFICIARIO'] > 0) & (df.loc[i, 'IDADE_APROX_BENEFICIARIO'] < 18):
        df.loc[i, 'FAIXA_ETA_BENEFICIARIO'] = 'Menos de 18 anos'
    
    elif (df.loc[i, 'IDADE_APROX_BENEFICIARIO'] >= 18) & (df.loc[i, 'IDADE_APROX_BENEFICIARIO'] < 24):
        df.loc[i, 'FAIXA_ETA_BENEFICIARIO'] = 'De 18 a 24 anos'
    
    elif (df.loc[i, 'IDADE_APROX_BENEFICIARIO'] >= 24) & (df.loc[i, 'IDADE_APROX_BENEFICIARIO'] < 30):
        df.loc[i, 'FAIXA_ETA_BENEFICIARIO'] = 'De 24 a 30 anos'
    
    elif (df.loc[i, 'IDADE_APROX_BENEFICIARIO'] >= 30) & (df.loc[i, 'IDADE_APROX_BENEFICIARIO'] < 36):
        df.loc[i, 'FAIXA_ETA_BENEFICIARIO'] = 'De 30 a 36 anos'
    
    elif (df.loc[i, 'IDADE_APROX_BENEFICIARIO'] >= 36) & (df.loc[i, 'IDADE_APROX_BENEFICIARIO'] < 48):
        df.loc[i, 'FAIXA_ETA_BENEFICIARIO'] = 'De 36 a 48 anos'
    
    elif (df.loc[i, 'IDADE_APROX_BENEFICIARIO'] >= 48) & (df.loc[i, 'IDADE_APROX_BENEFICIARIO'] < 60):
        df.loc[i, 'FAIXA_ETA_BENEFICIARIO'] = 'De 48 a 60 anos'
    
    else:
        df.loc[i, 'FAIXA_ETA_BENEFICIARIO'] = 'Mais de 60 anos'

In [27]:
df.sample()

ANO_CONCESSAO  CODIGO_IES                        NOME_IES  \
143005           2015          83  UNIVERSIDADE DO VALE DO ITAJAÍ   

            TIPO_BOLSA MODALIDADE_ENSINO            NOME_CURSO TURNO_CURSO  \
143005  BOLSA INTEGRAL        PRESENCIAL  Design De Interiores     Noturno   

       SEXO_BENEFICIARIO RACA_BENEFICIARIO  ANO_NASCIMENTO_BENEFICIARIO  \
143005         Masculino             Parda                         1997   

        IDADE_APROX_BENEFICIARIO FAIXA_ETA_BENEFICIARIO  \
143005                        18        De 18 a 24 anos   

       BENEFICIARIO_DEFICIENTE_FISICO REGIAO_BENEFICIARIO UF_BENEFICIARIO  
143005                            NÃO                 SUL              SC

# Analisando os dados

### Tipos de bolsas

In [28]:
df['TIPO_BOLSA'].value_counts()

BOLSA INTEGRAL       185086
BOLSA PARCIAL 50%     67564
Name: TIPO_BOLSA, dtype: int64

### Modalidade de ensino

In [29]:
df['MODALIDADE_ENSINO'].value_counts()

PRESENCIAL              203494
EDUCAÇÃO A DISTÂNCIA     49156
Name: MODALIDADE_ENSINO, dtype: int64

### Top 10 - Cursos

In [30]:
df['NOME_CURSO'].value_counts().head(10)

Administração                 28039
Direito                       20794
Pedagogia                     18519
Ciências Contábeis            14408
Engenharia Civil              11165
Educação Física                9488
Enfermagem                     8966
Gestão De Recursos Humanos     8696
Psicologia                     7201
Engenharia De Produção         6530
Name: NOME_CURSO, dtype: int64

### Turno predominante

In [31]:
df['TURNO_CURSO'].value_counts()

Noturno              141159
Curso a distância     49156
Matutino              48118
Integral               7426
Vespertino             6791
Name: TURNO_CURSO, dtype: int64

### Sexo dos beneficiários

In [32]:
df['SEXO_BENEFICIARIO'].value_counts()

Feminino     140407
Masculino    112243
Name: SEXO_BENEFICIARIO, dtype: int64

### Raça dos benenficiários

In [33]:
df['RACA_BENEFICIARIO'].value_counts()

Parda            110624
Branca           105265
Preta             32031
Amarela            4306
Não Informada       219
Indígena            205
Name: RACA_BENEFICIARIO, dtype: int64

### Faixa etária dos beneficiários

In [34]:
df['FAIXA_ETA_BENEFICIARIO'].value_counts()

De 18 a 24 anos     154463
De 24 a 30 anos      50132
De 30 a 36 anos      24323
De 36 a 48 anos      17410
De 48 a 60 anos       3120
Menos de 18 anos      3001
Mais de 60 anos        201
Name: FAIXA_ETA_BENEFICIARIO, dtype: int64

### Beneficiários deficientes físicos

In [35]:
df['BENEFICIARIO_DEFICIENTE_FISICO'].value_counts()

NÃO    250565
SIM      2085
Name: BENEFICIARIO_DEFICIENTE_FISICO, dtype: int64

### Região x Bolsas

In [36]:
df['REGIAO_BENEFICIARIO'].value_counts()

SUDESTE           115715
SUL                49627
NORDESTE           45947
CENTRO-OESTE       22848
NORTE              18502
Não Registrada        11
Name: REGIAO_BENEFICIARIO, dtype: int64

### Top 5 - UF's x Bolsas

In [37]:
df['UF_BENEFICIARIO'].value_counts().head()

SP    68489
MG    28038
RS    22332
PR    16914
RJ    14756
Name: UF_BENEFICIARIO, dtype: int64

### Top 10 - IE's x Bolsas

In [38]:
df['NOME_IES'].value_counts().head(10)

UNIVERSIDADE PITÁGORAS UNOPAR             14940
UNIVERSIDADE PAULISTA                     11569
UNIVERSIDADE ESTÁCIO DE SÁ                 7791
UNIVERSIDADE LUTERANA DO BRASIL            6089
CENTRO UNIVERSITÁRIO INTERNACIONAL         4728
UNIVERSIDADE ANHANGUERA - UNIDERP          4613
UNIVERSIDADE SÃO JUDAS TADEU               4486
CENTRO UNIVERSITÁRIO LEONARDO DA VINCI     4246
UNIVERSIDADE NOVE DE JULHO                 3900
CENTRO UNIVERSITÁRIO EURO-AMERICANO        3888
Name: NOME_IES, dtype: int64

## Criando um novo arquivo csv

In [39]:
novo_df = df

In [40]:
novo_df.to_csv('N_Datasets/Novo_Prouni2015.csv', sep=';', encoding='ISO-8859-1', index = False)